### ISSUE TITLE
Crash with Optifine when custom colors is enabled

### ISSUE THREAD

**Issue by Omnak** [TominoCZ/FancyBlockParticles#6]
If using a resource pack with Optifine where the resource pack does not provide custom colors, and Fancy Block particles is installed, some particles fail to render and Minecraft crashes. Steps to easily reproduce:

1) Install Fancy Block Particles and Optifine
2) Install a resource pack that does not provide custom colors (if you don't know of any, [SuikaCraft's continuation of Isabella](https://github.com/yurisuika/SuikaCraft) is a good candidate to test with).
3) Once in game, turn on custom colors using Optifine.
4) Build a nether portal and activate it
5) At this point, the game will have crashed. It does for me. Every time. Without fail.

I have several other mods installed as well, but this happens on a bare installation of just Forge, Optifine, and Fancy Block Particles. It also stops happening immediately after Fancy Block Particles is disabled, whether in game, or pre-launch.

Minecraft's crash log:

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  CreativePatchingLoader (CreativeCore v1.7.0 mc1.11.2.jar)
  LoadingPlugin (ChunkAnimator-MC1.11-1.2.jar)
  SPFMLLoadingPlugin (Sound-Physics-1.11.2.jar)
  ShoulderPlugin (ShoulderSurfing-1.11.2-1.4.jar)
  ItemPatchingLoader (ItemPhysic Lite 1.3.4 mc1.11.2.jar)
  DynamicSurroundingsCore (DynamicSurroundings-1.11.2-3.3.3.1.jar)
Contact their authors BEFORE contacting forge

// Ouch. That hurt :(

Time: 4/11/17 2:54 AM
Description: Loading screen debug info

This is just a prompt for computer specs to be printed. THIS IS NOT A ERROR


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.11.2
	Operating System: Linux (amd64) version 4.4.59-1-MANJARO
	Java Version: 1.8.0_121, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 668865976 bytes (637 MB) / 851443712 bytes (812 MB) up to 2505572352 bytes (2389 MB)
	JVM Flags: 2 total; -Xms128m -Xmx2688m
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: 
	Loaded coremods (and transformers): 
CreativePatchingLoader (CreativeCore v1.7.0 mc1.11.2.jar)
  
LoadingPlugin (ChunkAnimator-MC1.11-1.2.jar)
  lumien.chunkanimator.asm.ClassTransformer
SPFMLLoadingPlugin (Sound-Physics-1.11.2.jar)
  com.sonicether.soundphysics.SoundPhysicsCore
ShoulderPlugin (ShoulderSurfing-1.11.2-1.4.jar)
  com.teamderpy.shouldersurfing.asm.ShoulderTransformations
ItemPatchingLoader (ItemPhysic Lite 1.3.4 mc1.11.2.jar)
  com.creativemd.itemphysic.ItemTransformer
DynamicSurroundingsCore (DynamicSurroundings-1.11.2-3.3.3.1.jar)
  org.blockartistry.mod.DynSurround.asm.Transformer
	GL info: ' Vendor: 'X.Org' Version: '3.0 Mesa 17.0.2' Renderer: 'Gallium 0.4 on AMD RS780 (DRM 2.43.0 / 4.4.59-1-MANJARO, LLVM 3.9.1)'
[02:54:48] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Chunk Animator, FMLFileResourcePack:CreativeCore, FMLFileResourcePack:Custom Crosshair Mod, FMLFileResourcePack:Dynamic Surroundings, FMLFileResourcePack:Fancy Block Particles, FMLFileResourcePack:Waila, FMLFileResourcePack:Real First-Person Render, FMLFileResourcePack:Shoulder Surfing, FMLFileResourcePack:Sound Physics, FMLFileResourcePack:ToroHealth Damage Indicators, FMLFileResourcePack:Zyin's HUD, SE-Soundfix-v0.1.zip, [1.9.2]Dramatic Skys [v1.6].zip, Stevian_Villagers_1.3.zip, SuikaCraft 1.11.zip, Lithos-HD Font for 1.11x.zip
[02:54:50] [Sound Library Loader/INFO]: Starting up SoundSystem...
[02:54:50] [Thread-8/INFO]: Initializing LWJGL OpenAL
[02:54:50] [Thread-8/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[02:54:50] [Thread-8/INFO]: OpenAL initialized.
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Initializing Sound Physics...
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: EFX Extension recognized.
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 193 created
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 194 created
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 195 created
[02:54:50] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 196 created
[02:54:50] [Sound Library Loader/INFO]: Sound engine started
[02:54:55] [Client thread/INFO]: Created: 8x8 textures-atlas
[02:54:55] [Client thread/INFO]: [STDOUT]: [itemphysic] Patched net.minecraft.client.renderer.entity.RenderEntityItem ...
[02:54:59] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Chunk Animator, FMLFileResourcePack:CreativeCore, FMLFileResourcePack:Custom Crosshair Mod, FMLFileResourcePack:Dynamic Surroundings, FMLFileResourcePack:Fancy Block Particles, FMLFileResourcePack:Waila, FMLFileResourcePack:Real First-Person Render, FMLFileResourcePack:Shoulder Surfing, FMLFileResourcePack:Sound Physics, FMLFileResourcePack:ToroHealth Damage Indicators, FMLFileResourcePack:Zyin's HUD, LiteLoader, MnmUtils, ChatBubbles, VoxelMap, Gammabright, ChunkBorders, TabbyChat 2, SE-Soundfix-v0.1.zip, [1.9.2]Dramatic Skys [v1.6].zip, Stevian_Villagers_1.3.zip, SuikaCraft 1.11.zip, Lithos-HD Font for 1.11x.zip
[02:55:04] [Client thread/INFO]: Created: 1024x1024 textures-atlas
[02:55:07] [Client thread/WARN]: Skipping bad option: lastServer:
[02:55:08] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Chunk Animator, FMLFileResourcePack:CreativeCore, FMLFileResourcePack:Custom Crosshair Mod, FMLFileResourcePack:Dynamic Surroundings, FMLFileResourcePack:Fancy Block Particles, FMLFileResourcePack:Waila, FMLFileResourcePack:Real First-Person Render, FMLFileResourcePack:Shoulder Surfing, FMLFileResourcePack:Sound Physics, FMLFileResourcePack:ToroHealth Damage Indicators, FMLFileResourcePack:Zyin's HUD, LiteLoader, MnmUtils, ChatBubbles, VoxelMap, Gammabright, ChunkBorders, TabbyChat 2, SE-Soundfix-v0.1.zip, [1.9.2]Dramatic Skys [v1.6].zip, Stevian_Villagers_1.3.zip, SuikaCraft 1.11.zip, Lithos-HD Font for 1.11x.zip
[02:55:10] [Client thread/INFO]: Created: 1024x1024 textures-atlas
[02:55:19] [Client thread/INFO]: SoundSystem shutting down...
[02:55:19] [Client thread/WARN]: Author: Paul Lamb, www.paulscode.com
[02:55:19] [Sound Library Loader/INFO]: Starting up SoundSystem...
[02:55:19] [Thread-15/INFO]: Initializing LWJGL OpenAL
[02:55:19] [Thread-15/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[02:55:19] [Thread-15/INFO]: OpenAL initialized.
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Initializing Sound Physics...
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: EFX Extension recognized.
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 193 created
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 194 created
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 195 created
[02:55:20] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 196 created
[02:55:20] [Sound Library Loader/INFO]: Sound engine started
[02:55:45] [Client thread/INFO]: Realms library version == 1.10.16
[02:55:46] [MCO Compatability Checker #1/INFO]: Realms is available for this user
[02:56:04] [Client thread/INFO]: [STDOUT]: getting renderer
[02:56:04] [Client thread/INFO]: [STDOUT]: **Chat Bubble Renderer Initialized**
[02:56:04] [Client thread/INFO]: [STDOUT]: REALMS server detected!
[02:56:40] [Client thread/INFO]: Reloading ResourceManager: Default, FMLFileResourcePack:Forge Mod Loader, FMLFileResourcePack:Minecraft Forge, FMLFileResourcePack:Chunk Animator, FMLFileResourcePack:CreativeCore, FMLFileResourcePack:Custom Crosshair Mod, FMLFileResourcePack:Dynamic Surroundings, FMLFileResourcePack:Fancy Block Particles, FMLFileResourcePack:Waila, FMLFileResourcePack:Real First-Person Render, FMLFileResourcePack:Shoulder Surfing, FMLFileResourcePack:Sound Physics, FMLFileResourcePack:ToroHealth Damage Indicators, FMLFileResourcePack:Zyin's HUD, LiteLoader, MnmUtils, ChatBubbles, VoxelMap, Gammabright, ChunkBorders, TabbyChat 2, SE-Soundfix-v0.1.zip, [1.9.2]Dramatic Skys [v1.6].zip, PR Luminance 1.11.zip, Stevian_Villagers_1.3.zip, SuikaCraft 1.11.zip, Lithos-HD Font for 1.11x.zip
[02:56:40] [Client thread/WARN]: Missing sound for event: dsurround:missing
[02:56:40] [Client thread/INFO]: SoundSystem shutting down...
[02:56:40] [Client thread/WARN]: Author: Paul Lamb, www.paulscode.com
[02:56:40] [Sound Library Loader/INFO]: Starting up SoundSystem...
[02:56:41] [Thread-19/INFO]: Initializing LWJGL OpenAL
[02:56:41] [Thread-19/INFO]: (The LWJGL binding of OpenAL.  For more information, see http://www.lwjgl.org)
[02:56:41] [Thread-19/INFO]: OpenAL initialized.
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Initializing Sound Physics...
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: EFX Extension recognized.
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 193 created
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 194 created
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 195 created
[02:56:41] [Sound Library Loader/INFO]: [STDOUT]: [SOUND PHYSICS]: Aux slot 196 created
[02:56:41] [Sound Library Loader/INFO]: Sound engine started
[02:56:43] [Client thread/INFO]: Created: 1024x1024 textures-atlas
[03:02:39] [Client thread/FATAL]: Reported exception thrown!
net.minecraft.util.ReportedException: Exception while adding particle
	at net.minecraft.client.renderer.RenderGlobal.func_190570_a(RenderGlobal.java:2914) ~[bqm.class:?]
	at net.minecraft.client.renderer.RenderGlobal.func_180442_a(RenderGlobal.java:2887) ~[bqm.class:?]
	at net.minecraft.world.World.func_175720_a(World.java:1161) ~[ajs.class:?]
	at net.minecraft.world.World.func_175688_a(World.java:1140) ~[ajs.class:?]
	at net.minecraft.block.BlockPortal.func_180655_c(SourceFile:220) ~[apo.class:?]
	at net.minecraft.client.multiplayer.WorldClient.func_184153_a(WorldClient.java:454) ~[bnq.class:?]
	at net.minecraft.client.multiplayer.WorldClient.func_73029_E(WorldClient.java:442) ~[bnq.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1830) ~[bes.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1056) ~[bes.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:372) [bes.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:162) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:53) [NewLaunch.jar:?]
Caused by: java.lang.NullPointerException
	at CustomColors.updatePortalFX(CustomColors.java:1128) ~[CustomColors.class:?]
	at net.minecraft.client.renderer.RenderGlobal.func_190571_b(RenderGlobal.java:3024) ~[bqm.class:?]
	at net.minecraft.client.renderer.RenderGlobal.func_190570_a(RenderGlobal.java:2894) ~[bqm.class:?]
	... 24 more
```

Edit: replaced old crash report with more recent crash report.

**Comment by TominoCZ** [comment]
Do you have discord? Add me: #4919
I need a tester..

**Comment by Omnak** [comment]
I mean, I do have a Discord account, but due to either shyness or anti-socialness (depending on how you look at it), I'd prefer not to add anyone I don't know or don't need to add. I also don't think I'd make a great tester. I don't actually play Minecraft a whole lot, and I imagine testers need to be more reliable than that.

Is the information I provided in the OP not enough for you to figure out how to fix the problem, or at least to reproduce to get more information?

**Comment by Omnak** [comment]
Correction: Resource packs that do not have custom colors are not the problem. Resource packs that **do** have custom colors are the problem. And honestly that makes a whole lot more sense. I should have looked at Isabella's directory structure before creating this issue. After removing `/minecraft/resourcepacks/SuikaCraft 1.11.zip/assets/minecraft/mcpatcher/colormap` and `/minecraft/resourcepacks/SuikaCraft 1.11.zip/assets/minecraft/mcpatcher/color.properties`, I was able to run Minecraft perfectly fine with Fancy Block Particles and Optifine both installed, while custom colors is enabled.

In the case of Isabella, it just uses the default Minecraft color values anyway, so I'm not sure why those files even existed in the first place, and removing them does not make any difference on the way the resource pack looks. Maybe the maintainers are planning on adding custom colors in the future, which might be cool.

Basically, your mod is specifically incompatible with the `Custom Colors` feature of Optifine, but perfectly compatible with the rest of it.

**Comment by TominoCZ** [comment]
I mean I know that its the Custom Colors option doing it, but I have no idea why it crashes. I was testing the new version and so far it hasn't crashed yet but I need someone to test it.

**Comment by TominoCZ** [comment]
I will let this Issue open for now.
I've just released a new version of the mod and see if it works now :)

**Comment by Omnak** [comment]
After using the new version for about an hour with a few different resource packs that were previously crashing, I feel confident in saying that you've fixed the crash and that this issue can be closed.

Thanks! This is a great mod- the kind of subtle enhancement that would make a great addition to vanilla Minecraft.
